app/notification_email.go

```diff
@@ -284,6 +288,23 @@ func getFormattedPostTime(user *model.User, post *model.Post, useMilitaryTime bo
 	}
 }
 
+func (a *App) generateHyperlinkForChannels(postMessage, teamName, teamURL string) string {
+	r := regexp.MustCompile(`\~[-a-zA-Z0-9]+\b`)
```

Comment:
I don't love the regular expressions as a generic solution to find the channel strings, that string can match with other things, but because after that the channel is search and only modify the message when that happens, I think we are safe for a lot of corner cases.

I would add one check here: the `GetMessageForNotification` generate the message for "attachment-only" posts, would be nice to check it here before we try to do do any replacement. for not replace when is an attachment only notification.